### PR TITLE
Server unittests

### PR DESCRIPTION
**Pull request by pgcrooks**
This branch will stay open to track future unit test changes.

### CODE REVIEW

File: quiz/models.py [hunk 1]
```diff
@@ -51,15 +71,24 @@ def save(self, *args, **kwargs):
...
+    # Room name
     room = models.ForeignKey(Room, on_delete=models.CASCADE)
     # Handle i.e. username
     handle = models.CharField(max_length=200)
-    # user's score
+    # Score
     score = models.IntegerField(default=0)
     # has contestant completed
-    complete = models.IntegerField(default=1)
+    complete = models.BooleanField(default=False)
```
Comment (christopherkane) [inline quiz/models.py]: Where doth migration be?

File: quiz/models.py [hunk 2]
```diff
@@ -51,15 +71,24 @@ def save(self, *args, **kwargs):
...
 
     def __str__(self):
-        return self.handle
\ No newline at end of file
+        return self.handle
+
+    def clean(self):
```
Comment (christopherkane) [inline quiz/models.py]: Make sure Model clean does nothing

File: quiz/tests.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        """
+        question = Question()
+        with self.assertRaises(ValidationError):
+            question.full_clean()
+
+    def test_new_question(self):
+        """
+        Add a new question.
+        """
+        question = Question(question_text='foo')
```
Comment (christopherkane) [inline quiz/tests.py]: no answer test

File: quiz/tests.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        with self.assertRaises(ValidationError):
+            room.full_clean()
+
+    def test_room(self):
+        """
+        Add a room.
+        """
+        room = Room(name='test_room')
+        self.assertEqual(str(room), 'test_room')
+
```
Comment (christopherkane) [inline quiz/tests.py]: test code generation, try to duplicate and see save change code

File: quiz/models.py (hunk 1 above)
Comment (pgcrooks) [reply quiz/models.py]: Good shout, I'll add one this evening!

File: quiz/models.py (hunk 2 above)
Comment (pgcrooks) [reply quiz/models.py]: The `clean()` method is provided to allow custom validation, see https://docs.djangoproject.com/en/1.11/ref/models/instances/#django.db.models.Model.clean

File: quiz/tests.py (hunk 3 above)
Comment (pgcrooks) [reply quiz/tests.py]: I'll add one! It will fail for now until the model is updated to require answers.

File: quiz/tests.py (hunk 4 above)
Comment (pgcrooks) [reply quiz/tests.py]: Cool, will add!

**Comment by pgcrooks** [comment]
I'm pushing a failing test, so the Travis results are fine

**Comment by pgcrooks** [comment]
I referenced this thread to enforce lowercase Tag text: https://stackoverflow.com/questions/871037/django-overriding-init-for-custom-forms

**Pull request merged**